### PR TITLE
Fix multiple output version arbitrary config tests

### PR DESCRIPTION
**Pull request by gledis69**
With this small change, arbitrary config tests can have multiple acceptable correct outputs.

For an arbitrary config tests named `t`, now you can define `expected/t.out`, `expected/t_0.out`, `expected/t_1.out` etc and the test will succeed if the output of `sql/t.sql` is equal to any of the `t.out` or `t_{0, 1, ...}.out` files.

### CODE REVIEW

**Comment by gledis69** [comment]
Allows for the multi-output MERGE arbitrary config test in #6696 to pass.

**Comment by tejeswarm** [comment]
I'm testing these changes in my PR, will let you know if it worked.

**Comment by tejeswarm** [comment]
> I'm testing these changes in my PR, will let you know if it worked.

It's all good in my testing, approved.

File: src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py [hunk 1]
```diff
@@ -151,14 +149,19 @@ def copy_test_files_with_names(test_names, sql_dir_path, expected_dir_path, conf
...
 
         sql_name = os.path.join("./sql", test_name + ".sql")
-        output_name = os.path.join("./expected", test_name + ".out")
-
         shutil.copy(sql_name, sql_dir_path)
-        if os.path.isfile(output_name):
+
+        output_name = os.path.join("./expected", test_name + ".out")
+        alt_output_version_no = 0
+        while os.path.isfile(output_name):
```
Comment (tejeswarm) [inline src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py:156]: Please add a comment on why we do this i.e. significance of multiple output files and the need to check them.

Comment (tejeswarm) [reply src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py:156]: This will help new-comers.

File: src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py [hunk 2]
```diff
@@ -56,9 +56,7 @@ def run_for_config(config, lock, sql_schedule_name):
     )
     if config.user == cfg.REGULAR_USER_NAME:
         common.create_role(
-            config.bindir,
-            config.node_name_to_ports.values(),
-            config.user,
+            config.bindir, config.node_name_to_ports.values(), config.user
```
Comment (onderkalaci) [inline src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py:59]: unrelated change, could be nice to undo

**Review by onderkalaci — APPROVED** [review]
